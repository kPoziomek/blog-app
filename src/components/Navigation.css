.nav-container {
  display: flex;
  flex-direction: row-reverse;
  padding: 10px 10px;
  background-color: rgb(4, 69, 153);
}
.nav-link-box {
  display: flex;
  flex-direction: initial;
  text-decoration: none;
}
.nav-logo {
  width: 60%;
  display: flex;
  float: left;
  flex-direction: row;
  justify-content: space-between;
}
.logo {
  display: inherit;
}
.link-btn {
  text-decoration: none;
}
.nav-user-text {
  color: #fff;
  padding: 10px;
  margin: 0 auto;
}
.nav-img {
  width: 25px;
  height: 25px;
  padding: 10px;
  text-align: center;
  line-height: 25px;
}
.nav-text {
  color: #fff;

  width: 100px;
  padding: 10px;
  margin: 0 auto;
  text-align: center;
}

.nav-login-link {
  background-color: transparent;
  color: white;
  border: none;
  cursor: pointer;
  text-decoration: none;
  display: inline;
  margin: 0;
  padding: 0;
}
