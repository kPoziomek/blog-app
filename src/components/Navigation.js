import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useNavigate } from 'react-router-dom';
import { CardHeader, Button, Box } from '@mui/material';
import ReactLogo from '../Images/logo512.png';
import { authMe } from '../helpers/axiosConfig';

import './Navigation.css';

const Navigation = () => {
  const [isToken, setIsToken] = useState(false);
  const [userName, setUserName] = useState();
  const loggedIn = localStorage.key('token');

  useEffect(() => {
    if (loggedIn) {
      setIsToken(true);
      authMe().then((res) => {
        const userData = res.data;
        localStorage.setItem('tokenId', userData.id);
        console.log(userData);
        setUserName(`${userData.firstName} ${userData.lastName} `);
      });
    }
  }, [loggedIn]);

  const navigate = useNavigate();
  const handleLogin = () => {
    if (isToken) {
      setIsToken(false);
      localStorage.removeItem('token');
      localStorage.removeItem('tokenId');
      setUserName();
    } else {
      navigate('/login');
    }
  };
  return (
    <>
      <CardHeader
        className="nav-container"
        title={
          <Box className="nav-logo">
            <Link to="/" className="nav-link-box">
              <div className="logo">
                <img className="nav-img" src={ReactLogo} alt="logos" />
                <h3 className="nav-text">Blog-App</h3>
              </div>
            </Link>
            <div className="user">
              <h3 className="nav-user-text">
                {userName ? `Hello ${userName}` : 'Hello Stranger'}
              </h3>
            </div>
          </Box>
        }
        avatar={
          isToken ? (
            <div>
              <Link className="link-btn" to="/createArticle">
                <Button
                  className="nav-login-btn"
                  variant="contained"
                  color="primary"
                  sx={{ mx: 2 }}
                >
                  Create Article
                </Button>
              </Link>
              <Button
                className="nav-login-btn"
                variant="contained"
                color="primary"
                onClick={handleLogin}
              >
                logout
              </Button>
            </div>
          ) : (
            <Button
              className="nav-login-btn"
              variant="contained"
              color="primary"
              onClick={handleLogin}
            >
              Login
            </Button>
          )
        }
      />
    </>
  );
};

export default Navigation;
