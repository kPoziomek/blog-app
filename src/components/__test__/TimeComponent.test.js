import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom';

import Navigation from '../Navigation';

describe('Time Components Modal', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date('17 May 2022 09:00:00'));
  });
  afterAll(() => {
    jest.useRealTimers();
  });
  it('Clicking button open modal', async () => {
    const { baseElement } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    const button = screen.getByRole('button', { name: 'Open modal' });

    fireEvent.click(button);
    act(() => {
      jest.advanceTimersByTime(2000);
    });
    const today = new Date();
    const time = today.toTimeString();

    const modalTitle = screen.getByText(`Current Time is:`);
    const modalDescription = screen.getByText(`${time.slice(0, 8)}`);

    expect(baseElement).toMatchSnapshot();
    expect(modalTitle).toBeInTheDocument();
    expect(modalDescription).toBeInTheDocument();
  });

  it('Time stamp', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    const button = screen.queryByRole('button', { name: /open modal/i });

    fireEvent.click(button);
    act(() => {
      jest.advanceTimersByTime(2000);
    });
    const today = new Date();
    const time = today.toTimeString();
    const timer = screen.queryByText(`${time.slice(0, 8)}`);

    expect(timer).toBeInTheDocument();
  });
});
