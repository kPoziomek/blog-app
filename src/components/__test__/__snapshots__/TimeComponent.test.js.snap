// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Components Modal Clicking button open modal 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic nav-container css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular nav-toolbar css-1cndkqf-MuiToolbar-root"
      >
        <div
          class="nav-logo MuiBox-root css-0"
        >
          <a
            class="nav-link-box"
            href="/"
          >
            <div
              class="logo"
            >
              <img
                alt="logos"
                class="nav-img"
                src="logo512.png"
              />
              <h3
                class="nav-text"
                data-testid="nav-logo-text"
              >
                Blog-App
              </h3>
            </div>
          </a>
          <div
            class="nav-user-text"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedNeutral MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1tl8f8k-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Open modal
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="user"
          >
            <h3
              class="nav-user-text"
              data-testid="greeting-text"
            >
              Hello Stranger
            </h3>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <button
              aria-haspopup="true"
              aria-label="Account"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="menu-element"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-1azxqog-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="logo512.png"
                />
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    aria-describedby="modal-modal-description"
    aria-labelledby="modal-modal-title"
    class="MuiModal-root css-79ws1d-MuiModal-root"
    role="presentation"
    title="modal title"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-i9fmh8-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiBox-root css-1wnsr1i"
      tabindex="-1"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        id="modal-modal-title"
      >
        Current Time is:
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h3 css-1csm5re-MuiTypography-root"
        id="modal-modal-description"
      >
        09:00:02
      </h2>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="container MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 css-19fzd3z-MuiTypography-root"
        >
          Timer exercise
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h3 css-1csm5re-MuiTypography-root"
          id="modal-modal-description"
        >
          0
        </h2>
        <div
          class="mt-2 "
        >
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root btn mx-2  btn-primary css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            start
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-16hpjav-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            reset
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="container MuiBox-root css-0"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 css-19fzd3z-MuiTypography-root"
        >
          StopWatch Reducer
        </h2>
        <h2
          class="MuiTypography-root MuiTypography-h3 css-1csm5re-MuiTypography-root"
          id="modal-modal-description"
        >
          0
        </h2>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Start
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-16hpjav-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Stop
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Reset
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
